okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -263,18 +264,25 @@ public OkHttpClientStream newStream(MethodDescriptor<?, ?> method,
   @GuardedBy("lock")
   private void startStream(OkHttpClientStream stream) {
     Preconditions.checkState(stream.id() == null, "StreamId already assigned");
-    streams.put(nextStreamId, stream);
-    stream.start(nextStreamId);
+    int streamId = nextStreamId;
+    if (nextStreamId >= Integer.MAX_VALUE - 2) {
+      // Make sure nextStreamId is greater than all used ids, so that mayHaveCreatedStream()
+      // performs correctly.
+      nextStreamId = Integer.MAX_VALUE;
+    } else {
+      nextStreamId += 2;
+    }
+    streams.put(streamId, stream);
```

Comment:
This seems broken. Doesn't this actually create new streams after nextStreamId == MAX_VALUE? That means that we will then re-use MAX_VALUE for the streamId, which is _very bad_.